pageserver/src/tenant/timeline.rs

```diff
@@ -2422,6 +2429,11 @@ impl Timeline {
         let shard = self.get_shard_index();
         let this = self.myself.upgrade().expect("&self method holds the arc");
 
+        if let Some(ref index_part) = index_part {
+            self.last_aux_file_policy
+                .store(index_part.last_aux_file_policy());
+        }
+
```

Comment:
But haven't you just done that in the `Timeline::new` and are now repeating the same thing here?

Reply:
Only callsite for this method is `Tenant::timeline_init_and_sync`, which has before called `Timeline::new` via `Tenant::create_timeline_struct` (the best method name of all time).

`Tenant::create_timeline_struct` is also the only place where `Timeline::new` is called.

Reply:
Ah oki, now I can see so we need to read it again because of the duality of these settings. But it is not related to layer map loading so let's just move it out?

Reply:
fixed with https://github.com/neondatabase/neon/pull/7668/commits/344bc7a087f5c441c4c26b92f755d2f45c19aafc